e2e/test/scenarios/metrics/metrics-creation.cy.spec.ts

```diff
@@ -0,0 +1,254 @@
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import type { StructuredQuestionDetails } from "e2e/support/helpers";
+import {
+  createQuestion,
+  echartsContainer,
+  enterCustomColumnDetails,
+  getNotebookStep,
+  modal,
+  popover,
+  restore,
+} from "e2e/support/helpers";
+
+const { ORDERS_ID, PRODUCTS_ID } = SAMPLE_DATABASE;
+
+type QuestionDetails = StructuredQuestionDetails & { name: string };
+
+const ORDER_COUNT_DETAILS: QuestionDetails = {
+  name: "Orders metric",
+  type: "metric",
+  query: {
+    "source-table": ORDERS_ID,
+    aggregation: [["count"]],
+  },
+  display: "scalar",
+};
+
+const PRODUCT_COUNT_DETAILS: QuestionDetails = {
+  name: "Products metric",
+  type: "metric",
+  query: {
+    "source-table": PRODUCTS_ID,
+    aggregation: [["count"]],
+  },
+  display: "scalar",
+};
+
+describe("scenarios > metrics", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+    cy.intercept("POST", "/api/dataset").as("dataset");
+    cy.intercept("POST", "/api/card").as("createCard");
+  });
+
+  describe("data source", () => {
+    it("should create a metric for a table", () => {
+      cy.visit("/");
+      startNewMetric();
```

Comment:
Repeating this pattern in all tests will be slow.
WDYT we check the GUI flow in one test only, and you can create a new helper based on `startNewQuestion` that will immediately take you to the new metric page by generating the proper hash/link?

Reply:
Fixed